Lib/urllib/parse.py

```diff
@@ -497,9 +497,15 @@ def urlunsplit(components):
     This may result in a slightly different, but equivalent URL, if the URL that
     was parsed originally had unnecessary delimiters (for example, a ? with an
     empty query; the RFC states that these are equivalent)."""
-    scheme, netloc, url, query, fragment, _coerce_result = (
+    scheme, netloc, path, query, fragment, _coerce_result = (
                                           _coerce_args(*components))
-    if netloc or (scheme and scheme in uses_netloc and url[:2] != '//'):
+    if scheme in uses_netloc and path.startswith('//'):
```

Comment:
in practice this means someone passed in invalid components, but I agree that this is good desirable logic assuming uses_netloc is correct.  looking at uses_netloc, it contains the protocols anyone cares about and critically contains the empty string so this alone would fix the reported issue given the current code.

Reply:
I reverted the change to this line.  I'm not really sure what's it's trying to do so I decided better to not change it.  It seems like a weird edge case that doesn't make sense to me.  I.e. a scheme, not netloc and path that doesn't start with //.  E.g.
```
In [4]: urllib.parse.urlunsplit(('http', '', '/foo', '', ''))
Out[4]: 'http:///foo'
```
That doesn't seem right to me.  Maybe it makes sense to have scheme and not netloc for certain schemes?